Sources/PackageDescription/SupportedPlatforms.swift

```diff
@@ -71,6 +71,25 @@ public struct Platform: Equatable {
     public static let openbsd: Platform = Platform(name: "openbsd")
 }
 
+@available(_PackageDescription, introduced: 5.11)
+extension Platform: CaseIterable {
+    public static var allCases: [Platform] {
+        // Why visionOS does not need something like @available(_PackageDescription, introduced: 5.9)
```

Comment:
Some question here:
1. Why some new platform (eg. `Platform.visionOS`) is not marked with `@available(_PackageDescription, introduced: 5.x)` while some is marked(eg. `Platform.openbsd`)
2. We can't use `if #available(_PackageDescription 5.2, *)` currently. When a new platform is introduced on Swift 5.11 and is marked as `@available(_PackageDescription, introduced: 5.11)`, how can I append it conditionally?

Reply:
1. That seems like a mistake.
2. I think we would need `#available` support to make this work.

Reply:
1. Got it. Then maybe I can add check the time when they are added and add the corresponding available mark in this PR.
2. `if #avaiable(_PackageDescription 5.2, *)` is currently not a valid Swift statement and a Swift Proposal for this may be needed.

Reply:
I've seen some usage on Swift repo with `if #available(SwiftStdlib 5.3, *)` maybe it is already supported upstream 🤔 and all we should do is add some flag?

Reply:
```swift
let availabilityDefinition = PackageDescription.SwiftSetting.unsafeFlags([
    "-Xfrontend",
    "-define-availability",
    "-Xfrontend",
    "SwiftStdlib 5.7:macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0",
    "-Xfrontend",
    "-define-availability",
    "-Xfrontend",
    "SwiftStdlib 5.8:macOS 13.3, iOS 16.4, watchOS 9.4, tvOS 16.4",
    "-Xfrontend",
    "-define-availability",
    "-Xfrontend",
    "SwiftStdlib 5.9:macOS 9999, iOS 9999, watchOS 9999, tvOS 9999",
])
```

Found the trick here, looks like `SwiftStdlib` is just a typealis for the os check.